tracpro/polls/models.py

```diff
@@ -666,7 +671,46 @@ def category_counts(self):
         counts = Counter(categories)
         return counts.most_common()
 
+    def autocategorize(self):
+        """
+        Break down numeric answers into categories automatically, based somewhat
+        on ranges where there are bunches of responses.
+
+        See http://numpy.readthedocs.io/en/stable/reference/generated/numpy.histogram.html
+        where we're using the 'sqrt' bin assignment algorithm.
+
+        Silently ignores answers where `category` != "numeric", and any whose value
+        can't be successfully converted to a float.
+
+        Returns dictionary {
+          'categories': list of category names in order,
+          'data': list of counts in order
+        }
+
+        Category names are of the form "N.N-N.N".
+        """
+        answers = just_floats(self.filter(category='numeric').values_list('value', flat=True))
```

Comment:
I think what we actually want to filter on here is `self.filter(question__question_type=Question.TYPE_NUMERIC)`:

`answers = just_floats(self.filter(question__question_type=Question.TYPE_NUMERIC).values_list('value', flat=True))`